docs/developer/basic-form-support.md

```diff
@@ -0,0 +1,289 @@
+# Basic Form Support
+
+**NOTE:** This feature is under heavy development. Some of the described below may change in the future.
+
+Since Kubeapps 1.6.0, it's possible to attach a chart with a JSON schema that defines the structure of the `values.yaml` file. This JSON schema is used with two goals:
+
+ - Validate that the given values satisfy the schema defined. In case the submitted values are not valid, the installation or upgrade will fail.
+ - Present the user with a simple (basic) form so the chart is easier to deploy and configure.
+
+The goal of the basic form is to define the most common parameters that users typically modify before deploying a chart (like username and password) so they are able to modify them with a more user-friendly form.
+
+This document specifies what's needed to be defined in order to present this basic form to the users of a chart.
+
+## Create a values.schema.json
+
+This file, introduced with Helm v3, is a [JSON Schema](https://json-schema.org/) that defines the structure of the `values.yaml` file of the chart, including as many validations as needed. If a chart includes its schema, the values used are validated before submitting the new release.
+
+This file can define some or every possible value of the chart. Once it's written it should be included in the Helm package. The proposal to include it in Helm can be found [here](https://github.com/helm/helm/issues/5812).
+
+## Additional annotations used to identify basic parameters
+
+In order to identify, among all the values specified in the schema, the values that should be presented in the form, it's necessary to include some special tags.
+
+First of all, it's necessary to specify the tag `form`. This field should be unique and it should identify the parameter. All the properties marked with this key in the schema will be represented in the form. For example:
+
+```
+    "wordpressUsername": {
+      "type": "string",
+      "form": "username"
+    },
+```
+
+With the definition above, we are marking the value `wordpressUsername` as a value to be represented in the form (with the ID `username`). Note that we are also specifying the `type` that will be used to render the proper HTML components to represent the input in the form:
+
+![username-input](../img/username-input.png)
+
+In addition to the `type`, there are other tags that can be used to customize the way the parameter is represented:
+
+ - `title` is used to render the title of the parameter. If it's not specified, Kubeapps will use the ID of the component.
+ - `description` is used to include additional information of the parameter.
+ - `default` is used to set a default value. If it's not set, the default value from the `values.yaml` file will be used.
```

Comment:
Oh ... not sure that's a good idea? Why have two places to specify a default, so that you'll get different defaults depending on what UI you use? What is the reason for having a second default here rather than just using the one from the values.yaml always?

Reply:
sorry, I expressed it the other way. If the value is set in the `values.yaml` it will always be used as the default. In case that is not set and the `default` value is set in the schema, that one will be used.

Reply:
Yeah, but is there any reason to enable setting a default in the schema for a field at all, given the right place for that is already in the values.yaml? If not, I'd personally update this sentence removing any mention of default (or just saying the default is set in the values) and later remove support for a separate default in the schema. +1 either way, but I'm keen to know if there's a need at all for a second way to set the default for a field.